lib/SimpleHttpClient/Communicator.cpp

```diff
@@ -445,12 +445,10 @@ void Communicator::handleResult(CURL* handle, CURLcode rc) {
     (*rip->_options._curlRcFn)(rc);
   }
 
+  MUTEX_LOCKER(guard, _handlesLock);
   auto inProgress = _handlesInProgress.find(rip->_ticketId);
   auto curlHandle = (*inProgress).second->getSharedPtr();
```

Comment:
Just saw that there is no protection against not finding a specific ticket here.
So if the ticket is not found, we are dereferencing `inProgress` and refer to its `second` member, which should be undefined behavior. We should add a safety check for whether we have actually found the ticket or not here!